app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1354,15 +1399,72 @@ private void populatePlacesForAnotherLocation(
                 }));
     }
 
+    @SuppressLint("CheckResult")
+    public void loadPlacesDataAsync(List<Place> placeList, LatLng curLatLng) {
+        List<Place> places = new ArrayList<>(placeList);
+        int batchSize = 150;
```

Comment:
Judging from your screencast 150 seems to take a lot of time (actually I am getting `Could not load nearby places - Unexpected response code: 500`.), even in the countryside for 5 points it feels slow, so how about `3` for now? Thanks :-)